Objects/stringlib/find_max_char.h

```diff
@@ -20,23 +20,45 @@ Py_LOCAL_INLINE(Py_UCS4)
 STRINGLIB(find_max_char)(const STRINGLIB_CHAR *begin, const STRINGLIB_CHAR *end)
 {
     const unsigned char *p = (const unsigned char *) begin;
+    const unsigned char *_end = (const unsigned char *)end;
+    const size_t *aligned_end = (const size_t *)(_end - SIZEOF_SIZE_T);
```

Comment:
Just checking whether I understand correctly: `aligned_end` is not aligned, but good enough to serve as the end of the loop over aligned values. To make it really aligned we would need to do something like `aligned_end = _Py_SIZE_ROUND_DOWN(_end, SIZEOF_SIZE_T)`?

Reply:
Yes that is the gist of it. I tried something similar to _Py_SIZE_ROUND_DOWN and got segfaults so I opted for this simpler tried and true solution. The name `aligned_end` is probably not the best, but I can't think of a better name now.